mautic/mautic
# Support for emoji in emails2

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ❌ 
| New feature/enhancement? (use the a.x branch)      | ✅ 
| Deprecations?                          | ❌ 
| BC breaks? (use the c.x branch)        | ❌ 
| Automated tests included?              | ❌  
| Related user documentation PR URL      | N.A. 
| Related developer documentation PR URL | N.A. 
| Issue(s) addressed                     | N.A. 



#### Description:
The issue with email subject is that it is converted to a token and so fails with email_stats.tokens on send because that column is not utf8mb and thus does not support emojis natively.



#### Steps to reproduce the bug:
1. Execute query ALTER TABLE <prefix>email_stats CHANGE COLUMN tokens tokens LONGTEXT CHARACTER SET 'utf8' COLLATE 'utf8_unicode_ci' NULL DEFAULT NULL COMMENT '(DC2Type:array)'; to change the email_stats:token column's charset to utf8.
2. Create email with emoji in subject (Use this emoji: https://emojipedia.org/worm/)
3. It is observed that email is not received when sent (use campaign send email action to send emails)

#### Steps to test this PR:
1. Load up this PR
2. Revert the query above if it was executed using ALTER TABLE <prefix>email_stats CHANGE COLUMN tokens tokens LONGTEXT CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_ci' NULL DEFAULT NULL COMMENT '(DC2Type:array)'; to change charset to utf8mb4.
3. Follow steps to reproduce, email should be sent and received.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10844"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
